### PR TITLE
Template parts for header content

### PR DESCRIPTION
**Pull request by cjg89**
**Description**

- Set up template parts for headers, the site nav, and inner header contents to replace various markup-returning functions.
- Removed unnecessary `get_object_id()` and `get_object_field_id()` functions, and replaced their usage with just passing whatever `get_queried_object()` returns to `get_field()` when needed
- Moved media background functions out of functions.php and into their own dedicated file.  I'd like to eventually create template parts for these, but probably when we no longer have to rely on `set_query_var()` to pass along arguments.
- Added `get_header_type()` and `get_header_content_type()` functions for defining what header and header content template parts should be used for a given post/term.  The `get_header_content_title_subtitle` filter hook has been removed in favor of adjusting the returned header content type name in `get_header_content_type()` and creating a new corresponding template part.

**Motivation and Context**

To make it easier to define custom headers for upcoming researcher profiles, and for greater parity with the UCF WordPress Theme.

**How Has This Been Tested?**

Tested in Dev.

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Checklist:**

- [x] My code follows the code style of this project.
- [ ] My change requires an update to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: includes/header-functions.php [hunk 1]
```diff
@@ -136,229 +118,101 @@ function get_header_h1_option( $obj ) {
...
+ * @since 3.10.0
+ * @param mixed $obj A queried object (e.g. WP_Post, WP_Term), or null
+ * @param array $videos Array of video files set to use in the header background for the given $obj
+ * @param array $images Array of image files set to use in the header background for the given $obj
+ * @return string The header type name
+ */
+function get_header_type( $obj, $videos=null, $images=null ) {
+	$header_type = '';
+
+	$videos = ( $videos !== null ) ? $videos : get_header_videos( $obj );
```
Comment (jmbarne3) [inline includes/header-functions.php:135]: We should be able to replace these kinds of lines with the php7 shorthand at this point: `videos ?? get_header_videos ($obj )`. We using other php7 features elsewhere in plugins and what not, so we might as well take advantage.

File: includes/header-functions.php [hunk 2]
```diff
@@ -132,8 +132,8 @@ function get_header_h1_option( $obj ) {
 function get_header_type( $obj, $videos=null, $images=null ) {
 	$header_type = '';
 
-	$videos = ( $videos !== null ) ? $videos : get_header_videos( $obj );
-	$images = ( $images !== null ) ? $images : get_header_images( $obj );
+	$videos = $videos ?? get_header_videos( $obj );
+	$images = $videos ?? get_header_images( $obj );
```
Comment (jmbarne3) [inline includes/header-functions.php:136]:
```suggestion
	$images = $images ?? get_header_images( $obj );
```

**Review by jmbarne3 — CHANGES_REQUESTED** [review]
Looks like a small typo get inserted on the update.

File: includes/header-functions.php (hunk 2 above)
Comment (cjg89) [reply includes/header-functions.php]: 🤦‍♀️ thank you!

**Review by rjucf — APPROVED** [review]

**Pull request merged**